packages/frontend/src/components/shifts/ShiftBlock.tsx

```diff
@@ -117,8 +119,25 @@ function ShiftBlock(props: RootProps) {
     setDialogIsOpen(!dialogIsOpen);
   };
 
+  const handlePopperOpen = (event: React.MouseEvent<HTMLElement>) => {
+    if (!isEmptySlot) {
+      setAnchorEl(event.currentTarget);
+    }
+  };
+
+  const handlePopperClose = () => {
+    setAnchorEl(null);
+  };
```

Comment:
Both of these should probably be wrapped in `React.useCallback`... but not sure how much perf matters here

Reply:
I didn't even know `useCallback` was a thing. Good call! I used it where ever it made sense (i think lol).